File: InsuranceCompany.Core/Models/Question.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace InsuranceCompany.Core;

public partial class Question
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    public string? Text { get; set; }

    public bool? IsMandatory { get; set; }

    [ForeignKey(nameof(QuestionType))]
    public Guid? QuestionTypeId { get; set; }

    public virtual ICollection<AnswerValue> AnswerValues { get; } = new List<AnswerValue>();

    public virtual ICollection<Answer> Answers { get; } = new List<Answer>();

    public virtual ICollection<QuestionSurvey> QuestionSurveys { get; } = new List<QuestionSurvey>();

    public virtual QuestionType? QuestionType { get; set; }
}

File: InsuranceCompany.Client/src/app/models/Question.ts
import { InsuranceSurvey } from "./InsuranceSurvey";
import { QuestionType } from "./QuestionType";
import { Answer } from "./Answer";

export class Question {
    id : string;
    text : string;
    isMandatory : boolean;
    questionTypeId : string;
    insuranceSurvey : InsuranceSurvey;
    questionType :  QuestionType;
    answers :  Answer[];
    selectedAnswersForDelete: Answer[] = [];

    constructor(text: string) {
        this.text = text;
        this.answers = [];
        this.selectedAnswersForDelete = [];
    }
}